spirv_msl.cpp

```diff
@@ -8804,9 +8804,8 @@ string CompilerMSL::to_function_args(const TextureFunctionArguments &args, bool
 		{
 			forward = forward && should_forward(args.component);
 
-			if (const auto *var = maybe_get_backing_variable(img))
-				if (!image_is_comparison(get<SPIRType>(var->basetype), var->self))
-					farg_str += ", " + to_component_argument(args.component);
+			if (evaluate_constant_u32(args.component))
```

Comment:
This seems flawed if the application somehow uses gather y/z/w on a depth2d. Right now we would just end up using X (since there's nothing else we can do), but this would break compilation.

Reply:
Ah, I thought that's not a thing, do you mean a case where a gather-comparison is used on a multichannel texture? Would `image_is_comparison` check work?

Reply:
A texture2D might be used in a depth-compare context for certain scenarios, which forces depth2d type in Metal, but if applications then do textureGather(..., 1), this would break compilation since gather doesn't take component with depth2d in the first place.

Reply:
> With #1504 texture gathers do not have their component emitted even when they're not depth textures. I added a test for that and changed the condition to emit the component iff it's not x, which is the default.

I don't understand this. Are you saying there is a regression introduced by that PR?

Reply:
Yes, it fails the test I added here.

Reply:
Ah, ok. I see now. I introduced a regression ... I have an alternate fix. Would you be okay if I reused your test and committed it under a different PR?

Reply:
Sure, go ahead.